internal/model/disabled_users_map.go

```diff
@@ -0,0 +1,43 @@
+package model
+
+import (
+	"github.com/orcaman/concurrent-map"
+)
+
+// DisabledUsers is a type-safe and concurrent map keeping track of disabled users for idler.
+type DisabledUsers struct {
+	internal cmap.ConcurrentMap
```

Comment:
I wonder if it is possible to extend by just 

```go 
type DisabledUser struct {
  *cmap.Map
}
```
and adding methods to the new type. 

Have you tried doing this?

Reply:
@pradeepitm12 I am also thinking if the entire type is required? why not use cmap directly?

Reply:
Taken care in the new approach.